src/main/java/edu/hillel/homework/lesson13/filemanager/FileNavigator.java

```diff
@@ -0,0 +1,164 @@
+package edu.hillel.homework.lesson13.filemanager;
+
+import edu.hillel.homework.lesson13.filemanager.exceptions.InvalidFilePathException;
+import org.jetbrains.annotations.NotNull;
+
+import java.nio.file.FileAlreadyExistsException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+
+public class FileNavigator {
+
+    /**
+     * Map<String, List<FileData>> to store file path as key and List<FileData> as value
+     */
+    private final Map<String, List<FileData>> fileStorage;
+
+    /**
+     * File navigator initial constructor
+     */
+    public FileNavigator() {
+        this.fileStorage = new HashMap<>();
+    }
+
+    /**
+     * @param path     - adding file path for new FileData
+     * @param fileData - current FileData type
+     * @throws InvalidFilePathException   - throw when adding file path mismatch the actual file path
+     * @throws FileAlreadyExistsException - throw when such file name and file path already exist
+     */
+    public void add(@NotNull String path, @NotNull FileData fileData) throws
+            InvalidFilePathException,
+            FileAlreadyExistsException {
+        final String filePath = fileData.getFilePath();
+        final String fileName = fileData.getFileName();
+        if (path.equals(filePath)) {
+            if (!isFileAlreadyExist(fileData)) {
```

Comment:
I see it could be a big overhead using each time isFileAlreadyExist method and use foreach in map
What if use contains method before adding element to the map? If contains - just put file into list, if not - create a new list for this key and put element